gateway/src/apicast/configuration.lua

```diff
@@ -140,20 +145,44 @@ function _M.services_limit()
 end
 
 function _M.filter_services(services, subset)
-  subset = subset and util.to_hash(subset) or _M.services_limit()
-  if not subset or not next(subset) then return services end
-
   local s = {}
+  local serviceRegexpFilter  = env.value("APICAST_SERVICES_FILTER") or defaultRegexp
+
+  -- Checking that the regexp sent is correct, if not an empty service list
+  -- will be returned.
+  local compiled, _, _ = re_match_compile(serviceRegexpFilter, nil)
```

Comment:
This is pretty low-level API and it looks like the compiled regex should be destroyed (because it won't be GC'd?).

`ngx.re.match` returns a `captures, err`, so we can just use that as an indication the regex was invalid.

Maybe on an empty string? like `_, err = ngx.re.match('', filter, 'oj')` ?